### ISSUE TITLE
Amazing work! I encountered some issues while conducting my experiments.

### ISSUE THREAD

**Issue by zjj1105** [sporsho/3DUnOutDet#1]
Amazing work! Your paper is truly impressive and highly valuable, and I'm excited to see what you'll come up with in the future. Thank you so much for the open-source code! 
But I encountered some issues while conducting my experiments. 
1.Could you please clarify whether the mIOU in the paper represents the classification result for the snow category, or is it the average result for both snow and non-snow categories? In segmentation tasks, mIOU typically represents the multi-class average.
2. When testing the supervised version of 3D-OutDet, I obtained IOU results  close to the values reported in the paper, and the mIOU of the two categories is close to 95. However, in the unsupervised version, the IOU for the snow category is 79.05%, and the mIOU for the two categories  is close to 90%. Is there something specific I need to configure in the unsupervised code? Should I change the network depth from 1 to 2?

**Comment by sporsho** [comment]
Hi,
the IOU and mIOU in the paper represents snowy point classification (binary segmentation). But for the semantic segmentation, it is (as you said) the average across the class average. 
I just re-run the eval script and got 89.03% mIOU for snow detection with the default settings. See the attached screenshot. I am not sure what went wrong with your experiment. Have you made any modification to the existing source code?
![wads_3dunoutdet](https://github.com/user-attachments/assets/943bcc1f-fd0a-4d23-a623-573d3b662b2f)

**Comment by zjj1105** [comment]
Thank you for your reply. I didn't modify the parameter settings. I used sklearn to build the kdtree for neighbor point search because the environment setup for cuml is quite challenging. I tested the code on both the 2080ti and 4090 GPUs. I will carefully debug to identify where the issue lies. Your work is very interesting and comprehensive, and it is highly meaningful in addressing autonomous driving issues in extreme weather conditions. I will continue to follow your work and hope to stay in touch with you.
